src/rp2_common/pico_platform/include/pico/platform.h

```diff
@@ -233,7 +233,7 @@ extern "C" {
  *      int __force_inline my_function(int x) {
  *
  */
-#if defined(__GNUC__) && __GNUC__ <= 7
+#if defined(__GNUC__) && (__GNUC__ <= 6 || (__GNUC__ == 7 && (__GNUC_MINOR__ < 3 || !defined(__cplusplus))))
```

Comment:
So this is actually behaviour that changed in a "minor" version of GCC? Oh yuck :nauseated_face: